File: zd3.rb
import java.util.*

fun isPrime(n: Int): Boolean {
    if (n <= 1) {
        return false
    }
    var i = 2
    while (i * i <= n) {
        if (n % i == 0) {
            return false
        }
        i++
    }
    return true
}

fun main() {
    val numbers = mutableListOf<Int>()
    val scanner = Scanner(System.`in`)
    var num: Int

    // Ввод чисел
    println("Введите числа (для завершения введите отрицательное число):")
    num = scanner.nextInt()

    while (true) {
        if (num > 0) {
            num = scanner.nextInt()
            if (num < 0) {
                break
            }
            numbers.add(num)
        } else {
            println("Нужно вводить только числа:")
            return
        }
    }

    // Проверка чисел на простоту
    println("Результаты проверки чисел на простоту:")
    for (number in numbers) {
        if (isPrime(number)) {
            println("$number - простое число")
        } else {
            println("$number - не является простым числом")
        }
    }
}

File: zd3.java
import java.util.ArrayList;
import java.util.Scanner;

public class Main {
    
    public static boolean isPrime(int n) {
        if (n <= 1) {
            return false;
        }
        for (int i = 2; i * i <= n; ++i) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }
    
    public static void main(String[] args) {
        ArrayList<Integer> numbers = new ArrayList<>();
        Scanner scanner = new Scanner(System.in);
        int num;
        
         System.out.println("vvedite chislo (dla ostanovki luboe otricatelnoe chislo)");
        
        num = scanner.nextInt();
        
        while (true) {
            if (num > 0) {
                num = scanner.nextInt();
                if (num < 0) {
                    break;
                }
                numbers.add(num);
            } else {
                System.out.println("vvodite tolko chisla");
                return;
            }
        }
        
        
        System.out.println("rezultat");
        for (int number : numbers) {
            if (isPrime(number)) {
                System.out.println(number + " - prostoe");
            } else {
                System.out.println(number + " - not prostoe");
            }
        }
        
        scanner.close();
    }
}
